### PR TITLE
feat: drop distributed Mito2 table

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Drop distributed Mito2 table.

If "`engine = mito2`" is in the table info, drop the Mito2 table. Otherwise drop the original Mito table.

Note that since the Mito2 region is not able to be actually created on Datanodes now, there are no complete drop Mito2 table tests. I only mock the region server's grpc interfaces in the unit tests. Once the region server is ready on Datanode, it's easy to really test it.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

#2254

### CODE REVIEW

File: src/meta-srv/src/procedure/drop_table.rs [hunk 1]
```diff
@@ -145,11 +150,59 @@ impl DropTableProcedure {
...
+
+                    let request = region_request::Body::Drop(PbDropRegionRequest {
+                        region_id: region_id.as_u64(),
+                    });
+
+                    let client = clients.get_client(&datanode).await;
+                    let requester = RegionRequester::new(client);
+
+                    if let Err(err) = requester.handle(request).await {
+                        return Err(handle_request_datanode_error(datanode)(err));
```
Comment (WenyXu) [inline src/meta-srv/src/procedure/drop_table.rs:190]: Should we handle the datanode returns the `TableNotFound` error?

**Comment by MichaelScofield** [comment]
@WenyXu PTAL

**Review by WenyXu — APPROVED** [review]

File: src/meta-srv/src/procedure/create_table.rs [hunk 2]
```diff
@@ -197,15 +197,15 @@ impl CreateTableProcedure {
...
+                    create_region_request.region_id = region_id.as_u64();
+                    create_region_request.region_dir = region_dir(catalog, schema, region_id);
 
-                    PbRegionRequest::Create(create_table_request)
+                    PbRegionRequest::Create(create_region_request)
                 })
                 .collect::<Vec<_>>();
 
-            create_table_tasks.push(common_runtime::spawn_bg(async move {
+            create_region_tasks.push(common_runtime::spawn_bg(async move {
```
Comment (waynexia) [inline src/meta-srv/src/procedure/create_table.rs:208]: Why spawn here? Looks like the task won't be cancelled

Comment (MichaelScofield) [reply src/meta-srv/src/procedure/create_table.rs:208]: The spawned tasks are joined below, in line 221.

Comment (waynexia) [reply src/meta-srv/src/procedure/create_table.rs:208]: I see. But you can also join without spawning it

**Comment by MichaelScofield** [comment]
@waynexia PTAL

**Review by waynexia — APPROVED** [review]

**Pull request merged**